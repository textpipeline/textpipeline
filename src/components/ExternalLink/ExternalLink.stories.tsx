import { storiesOf } from '@storybook/react';
import React from 'react';

import ExternalLink from './ExternalLink';

storiesOf('ExternalLink', module).add('default', () => <ExternalLink href="/">Test Link</ExternalLink>);
