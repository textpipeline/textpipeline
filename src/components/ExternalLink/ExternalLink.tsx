import Link, { LinkProps } from '@material-ui/core/Link';
import React from 'react';

export interface ExternalLinkProps {
  readonly href: string;
  readonly className?: string;
  readonly color?: LinkProps['color'];
}

const ExternalLink: React.FC<ExternalLinkProps> = React.forwardRef<any, ExternalLinkProps>(
  ({ href, children, className, color }, ref) => (
    <Link href={href} rel="noreferrer" target="_blank" className={className} ref={ref} color={color}>
      {children}
    </Link>
  )
);

export default ExternalLink;
