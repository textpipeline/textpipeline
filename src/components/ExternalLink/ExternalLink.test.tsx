import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import ExternalLink from './ExternalLink';

describe('ExternalLink', () => {
  let rendered: ShallowWrapper;

  beforeEach(() => {
    rendered = shallow(<ExternalLink href="/" />);
  });

  it('renders correctly', () => {
    expect(rendered).toMatchSnapshot();
  });
});
