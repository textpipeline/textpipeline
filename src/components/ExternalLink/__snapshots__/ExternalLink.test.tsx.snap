// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalLink renders correctly 1`] = `
<WithStyles(ForwardRef(Link))
  href="/"
  rel="noreferrer"
  target="_blank"
/>
`;
